### ISSUE TITLE
[Documentation] Improve finnish rule documentation

### ISSUE THREAD

**Issue by C0ZEN** [cartant/eslint-plugin-rxjs#82]
**Context:**

Hello!
In my project, I use this configuration:

```yml
rxjs/finnish:
  - warn
  - functions: true
    methods: true
    names:
      ^canActivate$: false
      ^canActivateChild$: false
      ^canDeactivate$: false
      ^canLoad$: false
      ^intercept$: false
      ^resolve$: false
      ^validate$: false
      ^transform$: false
    parameters: true
    properties: true
    strict: true # Culprit
    types:
      ^EventEmitter$: false
      ^Store$: false
      ^Actions$: false
    variables: true
```

The `types` white-list was not working due to `strict` at `true`.  
Using it to `false` fixed my issue.

I have struggled to understand from the documentation what is the difference between using this option at `false` or at `true`.

**Suggestion:**

Could you add more details regarding this option in the documentation (possibly, add also some documentation for the other options)?
Thanks!

Link: https://github.com/cartant/eslint-plugin-rxjs/blob/main/docs/rules/finnish.md

**Comment by cartant** [comment]
IDK, the explanation is pretty clear, IMO:

> ... disallows the use of a `$` suffix with non-`Observable` identifiers

That's saying that if you use the suffix with types that aren't `Observable`s, it'll effect a failure. E.g.

```
const foo$ = "foo"; // Not allowed if strict is true
```

It sounds more like a bug, to me. Can you provide an example of how `types` isn't working when `strict` is `true`? I.e. a code snippet that would effect an unexpected rule failure?

**Comment by cartant** [comment]
Actually, I'll remove the negatives to make things simpler:

> ... a `strict` property that, if `true`, allows the `$` suffix to be used _only_ with identifiers that have an `Observable` type.

Of course, I still think this is likely to be a bug.

**Comment by C0ZEN** [comment]
@cartant sorry for the delay.
I checked this out once again, and I misunderstand the goal of this.
It's working as expected, there is no bug.
Sorry for the investigation and the tests you took, everything is fine!
